zebra-state/src/service/non_finalized_state/chain.rs

```diff
@@ -48,6 +48,29 @@ pub struct Chain {
     /// The configured network for this chain.
     network: Network,
 
+    /// The internal state of this chain.
+    inner: ChainInner,
+
+    // Diagnostics
+    //
+    /// The last height this chain forked at. Diagnostics only.
+    ///
+    /// This field is only used for metrics, it is not consensus-critical, and it is not checked
+    /// for equality.
+    ///
+    /// We keep the same last fork height in both sides of a clone, because every new block clones
+    /// a chain, even if it's just growing that chain.
+    pub(super) last_fork_height: Option<Height>,
+    // # Note
+    //
+    // Most diagnostics are implemented on the NonFinalizedState, rather than each chain.
+    // Some diagnostics only use the best chain, and others need to modify the Chain state,
+    // but that's difficult with `Arc<Chain>`s.
+}
+
+/// The internal state of [`Chain`].
+#[derive(PartialEq, Clone, Debug)]
```

Comment:
`Eq` is also guaranteed for this type, as long as all the types are `Eq` (which is checked by the derive).

https://doc.rust-lang.org/std/cmp/trait.Eq.html

```suggestion
#[derive(Clone, Debug, PartialEq, Eq)]
```